### DISCUSSION TITLE
How to install podman 4.4 on Ubuntu 22.04?

### DISCUSSION THREAD

**Discussion by electrofloat** [containers/podman#17362] [Q&A]
Ubuntu 22.04 is the latest LTS version of the OS. It only contains podman 3.4. 

The whatever repository here: https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04/amd64/
only has version 4.3. and have "(WIP - USE WITH CAUTION)" title with it.

This one: https://build.opensuse.org/project/show/devel:kubic:libcontainers:stable is a lot older.

The assets here: https://github.com/containers/podman/releases/tag/v4.4.0
only have .exe, .msi, .zip, something that has an extension .pkg, and podman-remote .tgz.

Where are the files to install podman 4.4 (not remote) on linux? I'd like to try out the new pasta driver for rootless containers.

**Comment by afbjorklund** [comment]
I think that is up to ubuntu, there are no binaries from podman upstream:
* https://github.com/containers/podman/issues/14065

There could be some Podman PPA or similar package repository available ?
* https://github.com/containers/podman/discussions/13097

**Reply by electrofloat** [reply]
Ubuntu LTS policy does not let them to update to version 4. It won't be updated in the Ubuntu repository.

Yeah, there should be at least a PPA available with Podman 4.4, but it seems podman maintainers are only interested in redhat enterprise linux.

**Reply by afbjorklund** [reply]
It will probably be updated for 24.04 LTS, it seems to be updated to 4.3.1 in debian "sid" at the moment

https://packages.debian.org/search?keywords=podman&searchon=names&exact=1&suite=all&section=all

**Reply by electrofloat** [reply]
That's for sure, but podman needs a way to have it installed on other linux distributions, like other than redhat enterprise linux.

**Reply by baude** [reply]
@electrofloat we rely on community support for distributions support.  The podman upstream team actually only builds for fedora.  every other distribution is done for others.  unfortunately, the upstream team does not have enough time/people/etc to perform builds on the side for each distribution.  this is not about RHEL or otherwise.  any chance you are a packager and would be willing to run a PPA or otherwise?

**Reply by electrofloat** [reply]
I see. I'm not a packager, but afaik anyone can have a ppa on ubuntu. The files to have it packaged as a .deb (the debian folder) must already exist since there is this repository here: https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04/amd64/.
I wish I had the time to work on these things, but I don't. I barely have enough to try out things and figure out why they don't work when I need them. Maybe if I live long enough and reach my retirement age, I'll have time for things like this.

**Reply by baude** [reply]
yeah, i think we are all in the same boat on that one.

**Reply by afbjorklund** [reply]
> ... other than redhat enterprise linux.

You can run it on Fedora, it is not only RHEL.

https://repology.org/project/podman/versions

**Reply by markkrj** [reply]
I think this is slowing podman adoption a lot. You guys should provide packages/repositories for, at least, Debian/Ubuntu and RHEL derived distros, like Docker do. You keep delivering nice features in podman, but sadly, it hits mainstream server distributions way too late. I, for one, went with Docker because of that.

**Reply by jetflux** [reply]
Same situation here, ubuntu package is just to outdated, and does'nt work properly. 
Had to go with docker because of this. but podman is so much better, leaner compared to docker.

**Reply by baude** [reply]
I would really love for you to communicate your wants to Ubuntu.  Is https://launchpad.net/ubuntu/+source/libpod and appropriate place?

**Comment by dtufood-kihen** [comment]
> I would really love for you to communicate your wants to Ubuntu. Is https://launchpad.net/ubuntu/+source/libpod and appropriate place?

I believe a more user friendly approach would be to create your own independent APT repository and distribute Ubuntu packages for Podman that way, similar to Docker: https://docs.docker.com/engine/install/ubuntu/

This would also allow you to replicate the same installation procedure, making it even easier to adapt to Podman.

**Reply by rhatdan** [reply]
Are you volunteering to maintain this?

**Reply by nerumo** [reply]
I understand your point of view, that maintaining all these packages for the different versions is quite some effort (effort that I'm also not willing/able to do :) ). But so many apps out there are only widespread, because they take care of the packaging.

 For me, I also can't recommend podman to my colleagues yet because there are fixes/features that are only on the latest versions.

**Reply by yhshin** [reply]
@rhatdan
You mentioned in your book `Podman in Action` that you advise to "get the packaged versions" due to complexity of meeting requirements to build all necessary tools such as `conmon`, etc.

Here the problem is that the available packaged versions are somewhat old versions especially the ones that included in LTS.  And it seems that any LTS like Ubuntu is not going to upgrade any apps in their system just because of users asked for it.  Especially if it includes a major version upgrade.

So all this looks like to me a situation that a company produced an amazing fabulous product but the only product that customer can get is slightly faded old ones while the shiny new ones that customers really want to get are displayed behind the show window.

The only way to get the new products seems to break the window.  All these methods available in the internet including the ones in here using `homebrew` (@lgonzalezsa), `kubic` (@electrofloat), and even building from source seems attempts to break the show window.

As a meager user, I just can only imagine a burden of packaging, but as users, @markkrj,  @dtufood-kihen, @beginner1010, @nerumo, and many others from other sites like SO pointed out, making packages available to users is very essential to widespread usage of this wonderful product.

P.S. Ubuntu 24.04 is released today and Podman 4.9.3 comes with it so this issue might seem over for now for users who would move to it.  But this issue will repeat after sometime later looking for a way to install podman 5 or 6 to Ubuntu 24.04.

**Reply by cgwalters** [reply]
In my case by far the biggest papercut is that the version of podman in `ubuntu-24.04` in GHA doesn't have heredoc support.

One thing I think we could all co-maintain is a GHA to update it - which is a more targeted thing?

**Reply by afbjorklund** [reply]
If you use `docker-buildx` (with podman) instead of `podman build`, then you can build with [newer Dockerfile syntax](https://www.docker.com/blog/introduction-to-heredocs-in-dockerfiles/)...

It would probably be a good idea for `podman buildx` to be configurable, in the same way that `podman compose` is?

**Comment by beginner1010** [comment]
@rhatdan @baude 
We switched from Docker to Podman as Kubernetes doesn't support Docker in Docker, owing to its non-compliance with the OCI standards. However, with recent versions, we've encountered issues, leading us to pin Podman to certain versions. The challenge arises because the OPENSUSE repository we use frequently updates and phases out the versions we rely on.

This isn't a formal request, but rather feedback: if there isn't a user-friendly method to install and maintain specific versions of Podman, we may have to consider alternative solutions. It's a reasonable expectation for users of any tool to have access to stable, version-specific installations. To not support this seems counterintuitive.

**Comment by fortuna** [comment]
You can download pre-built binaries from the releases page: https://github.com/containers/podman/releases

**Reply by computator** [reply]
That's just for Podman Remote, not the full Podman. It only has the full version for Mac and Windows.

**Reply by afbjorklund** [reply]
You can use `podman-remote-static` to create a VM on Linux (KVM), but it doesn't work with Podman Desktop.

**Reply by erlangparasu** [reply]
@afbjorklund any documentation about how to use these binaries?

**Reply by afbjorklund** [reply]
* https://github.com/podman-desktop/podman-desktop/discussions/5762

It actually "works" with Podman Desktop now, it will notice that the machine is there. But there is no GUI to actually create it.

There is also no documentation on how to install it, but I think that is on purpose. User is supposed to use the Linux distro.

* https://github.com/podman-desktop/podman-desktop/issues/6225

**Comment by pietrodantuono** [comment]
Maybe not ideal, but I figured that homebrew on Linux allows installing the latest podman version (5.0.1 as of today).

Tested on Zorin OS 17.1 Core.

https://formulae.brew.sh/formula/podman

**Reply by lgonzalezsa** [reply]
@pietrodantuono thanks for the pointer! using homebrew on Linux, tested podman 5.0.1 on Pop!OS Ubuntu LTS based 22.04 :muscle:

**Reply by afbjorklund** [reply]
Should probably be added to the list at https://podman.io/docs/installation, even if Homebrew is not a Linux distribution

```
brew install podman
```

**Reply by croqaz** [reply]
I'm on Ubuntu 22.04 so I had a very old version of Podman. Installed brew specifically to get the latest Podman, but I also installed Lima. The command `lima nerdctl` can be aliased as `docker` or `podman` and works perfectly. Using it for a few weeks.

**Reply by itsCheithanya** [reply]
> Should probably be added to the list at https://podman.io/docs/installation, even if Homebrew is not a Linux distribution
> 
> ```
> brew install podman
> ```
I want latest podman to be installed in system wide , it can't de done with brew right?,because i want to use podman go bindings and it gives this issue
![image](https://github.com/user-attachments/assets/f6d051a1-6219-47ca-a836-d1631563ccfa)
![image](https://github.com/user-attachments/assets/815c8446-a319-419a-af5e-8dde3c889fd9)

**Reply by afbjorklund** [reply]
Homebrew don't appear to install the systemd units, but is running the daemon with `podman system service --time=0`

**Comment by GodBleak** [comment]
If you want the latest version of Podman on the latest Ubuntu LTS, you could use https://software.opensuse.org/download/package?package=podman&project=home%3Aalvistack.

**Reply by lgonzalezsa** [reply]
Thanks @GodBleak I think I saw the alvistack in the past. I am curious about the software supply chain and how to make sure we use a trusted source. Do you any pointers on how alvistack takes care of the software supply chain ?

**Reply by itsCheithanya** [reply]
> If you want the latest version of Podman on the latest Ubuntu LTS, you could use https://software.opensuse.org/download/package?package=podman&project=home%3Aalvistack.

But im getting this netavark issue when installing using this method
![image](https://github.com/user-attachments/assets/fb2ceb49-0491-4765-b116-9a9e21435a84)
![image](https://github.com/user-attachments/assets/7a1cf418-ba1a-43f0-8e7e-dfd036dd861b)

**Reply by afbjorklund** [reply]
Sounds like a problem for https://build.opensuse.org/project/show/home:alvistack and https://github.com/alvistack ?

**Reply by ottopor** [reply]
running `sudo apt install podman-netavark`  sorted this issue on mine.

**Comment by boutell** [comment]
We might have better luck pleading with ubuntu to backport relevant bug fixes to the 3.x series they are releasing, in particular truly painful issues like:

```
Error: readlink /var/lib/containers/storage/overlay/l: invalid argument
```

See:

https://github.com/containers/podman/issues/8743

Rather than expecting them to support a new major version of podman on an existing LTS of Ubuntu.

**Reply by nickwilliams-zaxiom** [reply]
That's not the only painful issue. I'm having to switch back to Docker (having found no way to upgrade Podman on 22.04) due to [this known bug](https://bugs.launchpad.net/ubuntu/+source/libpod/+bug/2024394) that the Ubuntu maintainers have expressed no intention of fixing:

```
...
WARN[0000] Error validating CNI config file /home/williamsn/.config/cni/net.d/foobarname.conflist: [plugin firewall does not support config version "1.0.0"]
...
```

For me, Podman 3.4.4 is hosed and unusable.

**Reply by huyz** [reply]
@nickwilliams-zaxiom Apparently, all you have to do is modify the `*.conflist` files to say `0.4.0` instead.  A simple script (with `sed -i` or `perl -pi -e`) should do it (maybe even inserted into a podman/docker wrapper to make sure that the files are always fixed)

**Reply by nickwilliams-zaxiom** [reply]
I'm not sure that could work for me. I'm using `podman-compose`, so my networks are created when I `podman-compose up -d` and destroyed when I `podman-compose down`. There's never an opportunity to edit the transient network config file.

**Reply by huyz** [reply]
Would turning all the networks to external and pre-creating them be feasible?

**Reply by erlangparasu** [reply]
my workground is to create network first, than replace text 1.0.0 with 0.4.0, after that i create a contaienr using that modified network

**Comment by brooksvb** [comment]
Here I am trying to have Podman v5 on Ubuntu 24.04. It's such a bummer when I have to start adding unvetted 3rd party repositories or manually download and install packages just to have a recent software version. Really puts a cramp into my automation infrastructure.

**Reply by erlangparasu** [reply]
me too. i also try to setup podman-in-podman (to get latest version of podman running)

**Comment by boutell** [comment]
I'm in the same boat with you pretty much (Ubuntu 22.04 LTS in my case, so stuck on 3.x, and there have been no updates to correct some unpleasant bugs that don't exist in 4.x/5.x).

But maintaining ports for multiple distros is a lot of work, which is why teams with no dog at all in the distro fight still tend to abandon it and ship snaps etc. Snaps won't work for orchestration software like Podman, but from Red Hat's perspective they have done their part by creating and maintaining Podman.

Of course they offer packages for their own distros, but shouldn't the Debian community and/or Canonical worry about ports to that distro family?

That's why I think we'd be better off taking our concerns about backporting specific bug fixes to previous LTS releases to the Debian/Ubuntu maintenance teams and contributing patches.

Or running Docker, hey it's an option.

**Reply by brooksvb** [reply]
> Of course they offer packages for their own distros, but shouldn't the Debian community and/or Canonical worry about ports to that distro family?

Agreed. They should.

> Or running Docker, hey it's an option.

I'm totally on-board with rootless podman quadlet infrastructure. Once I've gotten through the subpar documentation and lots of gotchas, it's a clean and secure setup that's working well. I have a hard time believing that automating an up-to-date package is now my biggest headache in my project. I might consider running these servers on Fedora if it means up-to-date packages.

**Comment by k3wals** [comment]
The lack of official newer podman packages in 22.04 has been one of the biggest headaches for me. I've managed to get from 3.4.4 to 4.3.1 on 22.04 without hurdles and without using 3rd party sources. It's using packages from lunar (23.04) which is EOL though... Seems pretty stable from what I can see in my use.

I found this [gist](https://gist.github.com/shvchk/05ed652ada5a28d076a4d0471c2bb18d/) by shvchk, but I forked it to add a few more related packages. (crun and conmon)

https://gist.github.com/k3wals/84ed49a02cee61ab708a0c9730099363

**Reply by nickwilliams-zaxiom** [reply]
This may have worked three weeks ago, but it doesn't appear to anymore:

```
Ign:4 http://archive.ubuntu.com/ubuntu lunar InRelease
Hit:5 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:6 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease
Err:7 http://archive.ubuntu.com/ubuntu lunar Release
  404  Not Found [IP: 185.125.190.82 80]
Reading package lists... Done
E: The repository 'http://archive.ubuntu.com/ubuntu lunar Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

I haven't been able to find any way around it after about an hour of Googling. Sad. I guess I have to switch back to Docker. Podman is completely broken in Ubuntu 22.04 with a [known bug](https://bugs.launchpad.net/ubuntu/+source/libpod/+bug/2024394) that the Ubuntu maintainers seem unconcerned with fixing.

**Reply by k3wals** [reply]
Yeah, this was expected. I personally saved the *.deb packages in anticipation of this, but I'm not certain it'll satisfy all the dependencies.
```
buildah_1.28.2+ds1-1_amd64.deb
conmon_2.1.6+ds1-1_amd64.deb
crun_1.8-1_amd64.deb
golang-github-containers-common_0.50.1+ds1-4_all.deb
libsubid4_1%3a4.13+dfsg1-1ubuntu1_amd64.deb
netavark_1.4.0-3_amd64.deb
podman_4.3.1+ds1-5ubuntu1_amd64.deb
```
I'm upgrading to 24.04 just to be able to update Podman to at least 4.x.

**Reply by nickwilliams-zaxiom** [reply]
Yeah, I'd love to upgrade to 24.04. I'm trying to automate a test for a very niche edge case in our software, which I've decided is most easily done using containers, but we support about 40 platforms, so I have to have a solution that works in 20.04, 22.04, and 24.04.

**Comment by boutell** [comment]
I have reduced my rate of issues with Podman 3.x on Ubuntu 22.04LTS by eliminating the use of named volumes in favor of bind mounts. This removes Podman from a significant area of responsibility, it is no longer responsible for any persistent state. So far I haven't seen the dreaded overlayfs corruption errors again.

Probably my next step will be to migrate to plain old Docker, but I do not look forward to it as I already have logic to handle forwarding correctly to Podman's virtual network etc. in my firewall configuration.

**Comment by junminhong** [comment]
I recently compiled a guide on installing the latest version of Podman on Ubuntu 24.04, ensuring smooth operation. If you're interested, feel free to check it out:
[How to Install the Latest Podman on Ubuntu 24.04](https://library.wowkit.net/install-latest-podman-ubuntu-24-04#installation)

**Reply by afbjorklund** [reply]
When building for 20.04 and 22.04, you will need to compile also conmon and the networking (netavark/aardvark-dns, passt)

**Reply by junminhong** [reply]
Oh, thank you for letting me know.

**Comment by timur-g** [comment]
Thanks for this thread. I do not expect Canonical to update, they are limited to ppa:canonical-server/server-backports.
I was aware of Kubic and tried that to upgrade from 3.4.2 to 4.6.2, but later ended up with https://github.com/containers/podman/discussions/22991 which I could not resolve so reverted all.
New to me is Aalvistack. Updated to 5.5 but would update a myriad of other packages, does not seem like a good idea, better to upgrade OS.
As I do not really have need, I would not go the brew or compile way but good to see. Someone did similar at https://github.com/mgoltzsche/podman-static.

**Comment by boutell** [comment]
Another note for those who are reading this because they are experiencing:

```
Error: readlink /var/lib/containers/storage/overlay/l: invalid argument
```

... And similar errors:

In my case, these were occurring because of a scripted AWS `reboot-instances` CLI command that ended in a hard reboot, a situation older podman does not handle well.

Along with moving to Ubuntu 24.x, which at least provides podman 4.x (although that release is also beyond its end of life), removing this hard reboot from my provisioning scripts has pretty much eliminated the issue in practice. No more rebuilding containers.

But, I still strongly recommend my earlier advice: if you're nervous at all about podman filesystem integrity, just don't use named volumes, use bind mounts. Then the worst that can happen is you need to rebuild some containers, not the persistent data you can't just rebuild.

**Comment by erlangparasu** [comment]
already try to build from source but got some dependency error so i try:
https://github.com/mgoltzsche/podman-static
and it works for me

**Comment by KieranP** [comment]
Same situation here. Read an article about Podman over the weekend, gave it a go, worked great on my Mac. Went to use in production Ubuntu servers, and it installed a version from 1.5 years ago. We have strict OS update requirements as part of our SOC2 compliance, so if Ubuntu doesn't provide the latest Podman version, and the Podman team wont maintain their own PPA, then I simply cannot use Podman  :-( I hope this changes in the future.

**Reply by afbjorklund** [reply]
You can use old Podman on the old OS, or newer Podman on the new OS. That is the same as on RHEL 8 and 9.

If you want the new Podman, you need to build it from source (or change your OS). The OBS is discontinued.

**Reply by KieranP** [reply]
I am on the "new" OS though... Ubuntu 24.04 LTS is the most current recommended version for server infrastructure. I am sure most people would agree that using the LTS version, with 5 years of security updates, is a much better choice than a non-LTS version that only gets 9 months. I understand not wanting to support prior LTS releases or obscure linux distros, but the fact that the latest/recommended LTS of the most common server OS doesn't have up to date packages only hurts its adoption.

And yes, I could build it from scratch, and on a personal computer I would. But you can't scale that approach in production. Compiling podman on dozens or hundreds of servers, to then have to do it again in a couple of weeks after a security update, not to mentioned managing any issues that arise with outdated compile dependencies.... it's a security and maintenance nightmare.

So if I have to choose between a 1.5 year old Podman and an regularly updated Docker, for security and maintenance reasons, I'm going to need to choose Docker. Which is a shame, because I do like what Podman offers, and wish I could replace Docker with Podman.

**Reply by afbjorklund** [reply]
We had this discussion multiple times before (and we will again), but the position from the team was quite clear.

* https://github.com/containers/podman/discussions/25219

**Reply by Luap99** [reply]
> And yes, I could build it from scratch, and on a personal computer I would. But you can't scale that approach in production. Compiling podman on dozens or hundreds of servers, to then have to do it again in a couple of weeks after a security update, not to mentioned managing any issues that arise with outdated compile dependencies.... it's a security and maintenance nightmare.

You can build your own packages and provide your own repo to update all your server which is exactly what you want us to do, you could even provide the repos for everyone in the community to use to "fix" this issue for everyone here.

I can just repeat https://github.com/containers/podman/discussions/25219#discussioncomment-13168816

**Reply by boutell** [reply]
The upstream maintainers don't owe Debian or Ubuntu maintenance of older Podman versions beyond Podman's own lifecycle. That's something Debian and Ubuntu maintainers should do if they want to take it on.

**Reply by boutell** [reply]
The only secure alternative, to be honest, would be to remove podman from Debian and Ubuntu.

**Reply by boutell** [reply]
Debian does release backported security fixes for Node.js, another package which is "supported" in Debian much longer than it is supported upstream. It may be that Node.js critical security fixes are much easier to backport, I don't know.

**Reply by Luap99** [reply]
> Debian does release backported security fixes for Node.js, another package which is "supported" in Debian much longer than it is supported upstream. It may be that Node.js critical security fixes are much easier to backport, I don't know.

They also backport podman CVE fixes AFAIK, https://security-tracker.debian.org/tracker/source-package/podman
Each CVE will create a tracker there automatically. Now of course how fast it takes for them to patch these things can vary, the packer is a volunteer after all. And if you want to make this faster I am sure he would welcome any help on the official packaging.

**Reply by boutell** [reply]
Helpful to know how this is tracked. Right now there are just two CVEs listed, one hasn't been fixed yet (it's quite new), and the other didn't have to be fixed in then-current releases. So it's tough to gauge how things generally go from this page. (The usual disclaimer that these volunteer maintainers owe me nothing applies)

**Reply by Luap99** [reply]
I think the reason the tracker only shows two is because the package was named libpod before, https://security-tracker.debian.org/tracker/source-package/libpod

And note there are more CVE's that affected podman but they are tracked in the package where they appear, i.e. buildah https://security-tracker.debian.org/tracker/source-package/golang-github-containers-buildah

>  (The usual disclaimer that these volunteer maintainers owe me nothing applies)

Yes of course, just saying there are security processes involved on the debian side and they have a volunteer security team. It is not just the single podman packager who can patch it.

**Comment by boutell** [comment]
Good point. As long as it isn't made unduly difficult to find the fix
upstream, it's reasonable to expect people downstream to apply their own
patches, even if that's more difficult because they are choosing to pin to
an earlier major version etc.

On Tue, Sep 16, 2025 at 6:49 AM Paul Holzinger ***@***.***>
wrote:

> I think the reason the tracker only shows two is because the package was
> named libpod before,
> https://security-tracker.debian.org/tracker/source-package/libpod
>
> And note there are more CVE's that affected podman but they are tracked in
> the package where they appear, i.e. buildah
> https://security-tracker.debian.org/tracker/source-package/golang-github-containers-buildah
>
> (The usual disclaimer that these volunteer maintainers owe me nothing
> applies)
>
> Yes of course, just saying there are security processes involved on the
> debian side and they have a volunteer security team. It is not just the
> single podman packager who can patch it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/containers/podman/discussions/17362#discussioncomment-14413996>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27OXQ6M7CTWV3PE4MU33S7TNJAVCNFSM6AAAAAAURGSZIWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTINBRGM4TSNQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 

THOMAS BOUTELL | CHIEF EXECUTIVE OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
